nixos/modules/installer/sd-card/sd-image.nix

```diff
@@ -255,11 +264,8 @@ in
       '';
     }) {};
 
-    boot.postBootCommands = lib.mkIf config.sdImage.expandOnBoot ''
-      # On the first boot do some maintenance tasks
-      if [ -f /nix-path-registration ]; then
-        set -euo pipefail
-        set -x
+    boot.postBootCommands = let
+      expandOnBoot = if config.sdImage.expandOnBoot then ''
```

Comment:
```suggestion
      expandOnBoot = lib.optionalString config.sdImage.expandOnBoot ''
```

Reply:
thanks for the review :) fixed and fixed up